src/ui/components/SecondaryToolbox/react-devtools/suspense/hookNameCache.ts

```diff
@@ -0,0 +1,115 @@
+import { Location, MappedLocation } from "@replayio/protocol";
+import { createCache } from "suspense";
+
+import { updateMappedLocation } from "replay-next/src/suspense/PauseCache";
+import {
+  sourcesByIdCache,
+  streamingSourceContentsCache,
+} from "replay-next/src/suspense/SourcesCache";
+import { getPreferredLocation } from "replay-next/src/utils/sources";
+import { ReplayClientInterface } from "shared/client/types";
+import { locationToString } from "ui/actions/eventListeners/eventListenerUtils";
+
+const MAX_LINE_LENGTH = 200;
+
+export const hookNameCache = createCache<
+  [
+    replayClient: ReplayClientInterface,
+    locationOrMappedLocation: Location | MappedLocation,
+    fallbackName: string
+  ],
+  string | undefined
+>({
+  config: { immutable: true },
+  debugLabel: "hookNameCache",
+  getKey: ([_, locationOrMappedLocation, fallbackName]) => {
+    let locationString: string;
+    if (Array.isArray(locationOrMappedLocation)) {
+      locationString = locationOrMappedLocation.map(locationToString).join(",");
+    } else {
+      locationString = locationToString(locationOrMappedLocation);
+    }
+
+    return `${locationString}:${fallbackName}`;
+  },
+  load: async ([replayClient, locationOrMappedLocation, fallbackName]) => {
+    // If we already have a seemingly valid hook name,
+    // use it and skip the source fetching and parsing logic below.
+    if (fallbackName.startsWith("use")) {
+      return fallbackName.substring(3);
+    } else if (fallbackName.startsWith(".use")) {
+      // Some libraries (like React Router) assign hook displayNames with the format ".useRouter"
+      return fallbackName.substring(4);
+    }
+
+    let location: Location | undefined;
+    if (Array.isArray(locationOrMappedLocation)) {
+      const sourcesById = await sourcesByIdCache.readAsync(replayClient);
+
+      updateMappedLocation(sourcesById, locationOrMappedLocation);
+      location = getPreferredLocation(sourcesById, [], locationOrMappedLocation);
+    } else {
+      location = locationOrMappedLocation;
+    }
+
+    const { value: sourceContent } = await streamingSourceContentsCache.readAsync(
+      replayClient,
+      location.sourceId
+    );
+    if (!sourceContent) {
+      return fallbackName;
+    }
+
+    let lineIndex = location.line - 1;
+    let columnIndex = location.column;
+
+    // We could probably write a more efficient version of this that doesn't split the string if needed.
```

Comment:
:speech_balloon: yeah, I was running into a similar question while doing some of the ReactDOM source parsing work for the React Renders panel.

We could consider caching line-split sources separately as an intermediate option, to avoid re-splitting them for each location we are looking up.